src/template.py

```
@@ -0,0 +1,420 @@
+
+    @property
+    def mount_points(self):
+        return {}
+
+    def wait_for_port(self, port_num, waitclose=False, timeout=180):
+        starttime = time.time()
+
+        try:
+            self.docker_execute("which nc")
+        except subprocess.CalledProcessError:
+            raise Exception("nc not found in container")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)